File: code-awesome/go-gin-react/webshop/backend/src/pages/product..go
package pages

import (
	"fmt"
	"net/http"
	"strconv"
	"strings"

	"github.com/gin-gonic/gin"
)

type ProductPage struct {
	productID int
}

func NewProductPage(productID int) *ProductPage {
	return &ProductPage{productID: productID}
}

func (p *ProductPage) GetProductDetails() gin.HandlerFunc {
	return func(c *gin.Context) {
		// Simulate fetching product details from a database.
		productDetails := p.getProductDetailsFromDatabase()
		if productDetails == nil {
			c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to retrieve product details"})
			return
		}

		c.JSON(http.StatusOK, productDetails)
	}
}

func (p *ProductPage) getProductDetailsFromDatabase() map[string]interface{} {
	// Simulate a database query.  This is where a content injection vulnerability could exist.
	// Example:  If the productID comes from user input, and isn't properly sanitized,
	// an attacker could inject malicious SQL code.
	if p.productID < 1 {
		return nil
	}

	productDetails := map[string]interface{}{
		"product_id":       p.productID,
		"product_name":     "Cute Kitten",
		"price":            39.99,
		"description":      "A fluffy and adorable kitten.",
		"image_url":        "https://example.com/kitten.jpg",
		"stock_quantity": 100,
	}
	return productDetails
}

func ProductPageHandler(c *gin.Context) {
	productID := 1
	productPage := NewProductPage(productID)
	productPage.GetProductDetails()(c)
}

File: code-awesome/java-quarkus-vue/webshop/backend/src/pages/product..java
package pages;

import java.util.HashMap;
import java.util.Map;

public class product {

    private Map<String, String> productDetails = new HashMap<>();

    public product(String productName, String productDescription, double productPrice) {
        this.productDetails.put("productName", productName);
        this.productDetails.put("productDescription", productDescription);
        this.productDetails.put("productPrice", String.valueOf(productPrice));
    }

    public Map<String, String> getProductDetails() {
        return productDetails;
    }

    public static void main(String[] args) {
        product product1 = new product("Cute Kitten", "A fluffy, adorable kitten", 25.00);
        Map<String, String> details = product1.getProductDetails();
        System.out.println("Product Name: " + details.get("productName"));
        System.out.println("Product Description: " + details.get("productDescription"));
        System.out.println("Product Price: " + details.get("productPrice"));
    }
}